File: StoreWebApi/store-web-ui/src/app/core/models/customer.ts
import { Product } from "./product";

export type Customer = {
    id: number;
    firstName: string;
    lastName: string;
}

export interface CustomerRequest extends Customer {

}

export type CustomerPurchases = {
    purchaseId: number;
    customerId: number;
    customer: Customer;
    productId: number;
    product: Product;
}

export type CustomerOrder = {
    customerId: number;
    productIds: number[]
}

File: StoreWebApi/Models/ViewModel/Customer.cs
namespace StoreWebApi.Models.ViewModel
{
    public class CustomerOrderRequest
    {
        public int CustomerId {  get; set; }
        public required List<int> ProductIds { get; set; }
    }

    public class CustomerViewModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
    }

    public class CustomerRequest: CustomerViewModel
    {
    }
}
